gui/static/js/blocks/custom_blocks.js

```diff
@@ -78,4 +78,202 @@ Blockly.Python['wait'] = function(block) {
+
+  return [fcn+number_percent+")", Blockly.Python.ORDER_NONE];
+};
+
+Blockly.Blocks['turn_power_time'] = {
+  init: function() {
+    this.jsonInit(miniblocks.turn_power_time);
+  }
+};
+
+Blockly.Python['turn_power_time'] = function(block) {
+  var dropdown_direction = block.getFieldValue('direction');
+  var number_percent = block.getFieldValue('percent');
+  var number_seconds = block.getFieldValue('seconds');
+  // TODO: Assemble Python into code variable.
+  var fcn = {
+    turn_clockwise: "turn_clockwise(",
+    turn_counter_clockwise: "turn_counter_clockwise("
+  }[dropdown_direction];
+
+  var wait_cmd = "wait("+ number_seconds +")";
+
+  return [fcn+number_percent+")\n" +wait_cmd, Blockly.Python.ORDER_NONE];
+};
+
+// ================ WAIT BLOCK ================ //
+
+Blockly.Blocks['wait_seconds'] = {
+  init: function() {
+    this.jsonInit(miniblocks.wait_seconds);
+  }
+};
+
+Blockly.Python['wait_seconds'] = function(block) {
+  var number_seconds = block.getFieldValue('seconds');
+  // TODO: Assemble Python into code variable.
+  var wait_cmd = "wait("+ number_seconds +")";
+  return wait_cmd;
+};
+
+// ================ COMMUNICATION BLOCKS ================ //
+
+Blockly.Blocks['send_commands'] = {
+  init: function() {
+    this.jsonInit(miniblocks.send_commands);
+  }
+};
+
+Blockly.Python['send_commands'] = function(block) {
+  var dropdown_bot_name = block.getFieldValue('bot_name');
+  var statements_send_commands = Blockly.Python.statementToCode(block, 'send_commands');
+  // TODO: Assemble Python into code variable.
+  var code = 'pass #TODO Implement backend for send_commands';
+  return code;
+};
+
+Blockly.Blocks['wait_for_commands'] = {
+  init: function() {
+    this.jsonInit(miniblocks.wait_for_commands);
+  }
+};
+
+Blockly.Python['wait_for_commands'] = function(block) {
+  var dropdown_bot_name = block.getFieldValue('bot_name');
+  // TODO: Assemble Python into code variable.
+  var code = 'pass #TODO Implement backend for wait_for_commands';
+  return code;
+};
+
+Blockly.Blocks['while_wait_for_commands'] = {
+  init: function() {
+    this.jsonInit(miniblocks.while_wait_for_commands);
+  }
+};
+
+Blockly.Python['while_wait_for_commands'] = function(block) {
+  var dropdown_bot_name = block.getFieldValue('bot_name');
+  var statements_wait_commands = Blockly.Python.statementToCode(block, 'wait_commands');
+  // TODO: Assemble Python into code variable.
+  var code = 'pass #TODO Implement backend for while_wait_for_commands';
+  return code;
+};
+
+// ================ COLOR SENSING BLOCKS ================ //
+
+Blockly.Blocks['sees_color'] = {
+  init: function() {
+    this.jsonInit(miniblocks.sees_color);
+  }
+};
+
+Blockly.Python['sees_color'] = function(block) {
+  var dropdown_sensor_name = block.getFieldValue('sensor_name');
+  var dropdown_color_name = block.getFieldValue('color_name');
+  // TODO: Assemble Python into code variable.
+  var code = 'pass #TODO Implement backend for sees_color';
```

Comment:
The functions are already written (for the most part), so you should include those function calls instead of `pass`. Even if it's not fully tested, at least it generates what seems like syntactically correct code.